### PR TITLE
Fixed LLtoUTMUPSObject northing values in southern hemisphere.

### PR DESCRIPTION
**Pull request by BernardIgiri**
Northing values in the southern hemisphere are now correctly converted from Lat Lon to UTM. The unit tests have been updated to check northing values. In addition a isInUPSSpace convenience method is added with unit tests to help library users verify if a Lat Lon coordinate can be converted to UPS or not.

### CODE REVIEW

File: tests/tests.js [hunk 1]
```diff
@@ -1442,19 +1453,25 @@ describe('UPS Conversions', () => {
   describe('LLtoUTMUPS', () => {
     describe('convert to UTM when necessary', () => {
       it('80S', ()=> {
-        const utm = converter.LLtoUTMUPS(-80, 0)
-        const expected = "31 441"
-        chai.expect(utm.substr(0, expected.length)).to.equal(expected)
+        const utm = converter.LLtoUTMUPS(-80, 0).split(" ")
+        const expected = ["31", "441", "111"]
+        expected.forEach((_, i) => {
+          chai.expect(utm[i].substr(0, expected[i].length)).to.equal(expected[i])
```
Comment (Variadicism) [inline tests/tests.js]: The `substr` calls here discard proper checks for values that have the prefix of the expected numbers, e.g., if the UTM returned was "`31C 4415mE 1116mN`". Would a regex match work? The above could be matched against `/31[A-Z] 441mE 111mN/`.

File: tests/tests.js [hunk 2]
```diff
@@ -810,6 +810,17 @@ describe('Convert Lat/Lon to USNG', function(){
...
 describe('UPS Conversions', () => {
+  describe('isInUPSSpace', () => {
+    it('North pole', () => {
+      chai.expect(converter.isInUPSSpace(85)).to.equal(true)
+    })
+    it('South pole', () => {
+      chai.expect(converter.isInUPSSpace(-81)).to.equal(true)
+    })
+    it('Not UPS Space', () => {
+      chai.expect(converter.isInUPSSpace(35)).to.equal(false)
```
Comment (Variadicism) [inline tests/tests.js]: :pencil2: Just to be sure, could you also add a test to ensure that `false` is returned for a negative values greater than -80?

File: tests/tests.js [hunk 3]
```diff
@@ -1453,25 +1460,19 @@ describe('UPS Conversions', () => {
   describe('LLtoUTMUPS', () => {
     describe('convert to UTM when necessary', () => {
       it('80S', ()=> {
-        const utm = converter.LLtoUTMUPS(-80, 0).split(" ")
-        const expected = ["31", "441", "111"]
-        expected.forEach((_, i) => {
-          chai.expect(utm[i].substr(0, expected[i].length)).to.equal(expected[i])
-        })
+        const utm = converter.LLtoUTMUPS(-80, 0)
+        const expected = /^31[C]? 441[0-9]{3,3}mE 1116[0-9]{3,3}mN$/
```
Comment (Variadicism) [inline tests/tests.js]: `31[C]?` can be simplified to `31C?`.

Comment (Variadicism) [reply tests/tests.js]: `{3,3}` can be simplified to `{3}`.

Comment (Variadicism) [reply tests/tests.js]: :pencil2: `[0-9]` can be replaced with `\d`.

**Comment by rymach** [comment]
Hero successful 🍾

**Comment by BernardIgiri** [comment]
USNG4J equivalent PR: https://github.com/codice/usng4j/pull/15

File: package.json [hunk 4]
```diff
@@ -53,5 +53,8 @@
     "format": "tsfmt -r"
   },
   "author": "codice",
-  "license": "MIT"
+  "license": "MIT",
+  "dependencies": {
+    "chai-match": "^1.1.1"
```
Comment (Lambeaux) [inline package.json]: ❓ Should this be under `"devDependencies"` instead? Do we need this at runtime or only to execute tests?

Comment (dimitry-sherman) [reply package.json]: Only tests

File: src/usng.ts [hunk 5]
```diff
@@ -638,15 +638,27 @@ extend(Converter.prototype, {
...
-    // Constrain reporting UTM coords to the latitude range [80S .. 84N]
-    return (lat > 84.0 || lat < -80.0)
-      ? { ...this.LLtoUPS(lat, lon), zoneNumber: 0 }
-      : { ...this.LLtoUTM(lat, lon), northPole: lat >= 0 };
+    if (this.isInUPSSpace(lat)) {
+      return { ...this.LLtoUPS(lat, lon), zoneNumber: 0 }
+    } else {
+      let utmcoords = [0, 0, 0, 'N']
+      this.LLtoUTMwithNS(lat, lon, utmcoords)
+      return {
```
Comment (andrewkfiedler) [inline src/usng.ts]: This confused me a bit at first because I didn't realize utmcoords would be mutated by LLtoUTMwithNS.  

Probably my own preference here, but I would prefer to avoid the mutation and have LLtoUTMwithNS return something instead.  

That's outside this change though.

File: src/usng.ts [hunk 6]
```diff
@@ -638,15 +638,27 @@ extend(Converter.prototype, {
     }
   },
 
+  isInUPSSpace(lat) {
```
Comment (andrewkfiedler) [inline src/usng.ts]: Since this is a typescript file we could define the type for this function to catch possible issues at compile time.
